### PR TITLE
Drag node edge callback

### PR DESCRIPTION
**Pull request by jweinstein92**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Error is thrown when a dragNode or dragEdge callback is used for overriding the drag.
Issue Number: N/A

## What is the new behavior?
Remove `dragEdge` callback option. Get the node props of the dragging node to use with the `dragNode` callback function.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/Canvas.tsx [hunk 1]
```diff
@@ -319,13 +321,23 @@ const InternalCanvas: FC<CanvasProps & { ref?: Ref<CanvasRef> }> = forwardRef(
     );
 
     useEffect(() => {
-      if (dragNodeData.children) {
+      if (dragNodeData) {
```
Comment (amcdnl) [inline src/Canvas.tsx:324]: Putting elements in state doesn't feel like a good idea.

File: stories/Drag.stories.tsx [hunk 2]
```diff
@@ -465,6 +465,7 @@ export const NestedNodeRearranging = () => {
         nodes={nodes}
         edges={edges}
         node={<Node dragType="node" />}
+        dragNode={(node: NodeProps) => (<Node {...node} dragType="node" />)}
```
Comment (amcdnl) [inline stories/Drag.stories.tsx:468]: Why do you need this callback?

Comment (jweinstein92) [reply stories/Drag.stories.tsx:468]: you don't. it was just a test to make sure that having a callback for `dragNode` works

File: src/Canvas.tsx (hunk 1 above)
Comment (jweinstein92) [reply src/Canvas.tsx:324]: There needs to be a way to identify and dynamically generate the JSX element based on the `dragNode` callback. Do you have any recommendations on a better approach?

File: stories/Drag.stories.tsx (hunk 2 above)
Comment (amcdnl) [reply stories/Drag.stories.tsx:468]: OK - lets make these demos as simple as possible - let's just use the nromal one

File: src/Canvas.tsx (hunk 1 above)
Comment (jweinstein92) [reply src/Canvas.tsx:324]: I think I have a good solution. Changes incoming

**Review by amcdnl — APPROVED** [review]

**Pull request merged**